t/26-video_stream.t

```diff
@@ -205,6 +205,24 @@ subtest 'input events' => sub {
         "ctrl-x\n",
         "s\n", "o\n", "m\n", "e\n", "spc\n", "t\n", "e\n", "s\n", "t\n", "ret\n",
     ], "correct commands sent";
+
+    $bmwqemu::vars{GENERAL_HW_KEYBOARD_URL} = 'http://127.0.0.42:42000/cmd';
+
+    # mock ua
+    my $i = 0;
+    my $user_agent_mock = Test::MockModule->new('Mojo::UserAgent');
+    my $http = Test::MockModule->new('Mojo::Transaction::HTTP');
+    my $req_mock = Test::MockModule->new('Mojo::Message::Request');
+    $user_agent_mock->redefine(start => sub ($ua, $tx) { });
+    $user_agent_mock->redefine(get => sub { Mojo::Transaction::HTTP->new });
+    $http->redefine(result => sub { $i = $i + 1; Mojo::Message::Response->new->code(200)->body("hallo") });
+
+    $console->activate;
+    $console->send_key({key => 'a'});
+    $console->send_key({key => 'ctrl-x'});
+    $console->type_string({text => "some test\n"});
+    $console->disable;
+    is $i, 3, 'right number of http requests called';
```

Comment:
You could also check whether the right requests have been made, e.g. you could push the attempted requests in an array in your mocking (`$tx->req` should return a https://docs.mojolicious.org/Mojo/Message/Request object).